src/plugins/NEMD/DriftCtrl.cpp

```diff
@@ -228,63 +249,45 @@ void DriftCtrl::beforeForces(ParticleContainer* particleContainer, DomainDecompB
 				continue;
 			
 			// check if target component
-			uint32_t cid_zb = it->componentid();
-			uint32_t cid_ub = cid_zb+1;
-			if(cid_ub != _target.cid)
-				continue;
+			uint32_t cid_ub = 0;
+			if (_target.cid != 0) {
+				cid_ub = it->componentid() + 1;
+				if (cid_ub != _target.cid) {
+					continue;
+				}
+			}
 			
 			uint32_t yPosID = floor( (yPos-_range.yl) / _range.subdivision.binWidth.actual);
-
-			//~ it->setv(0, it->v(0) + _sampling.at(cid_ub).mom_corr.at(0).at(yPosID) ); 
-			it->setv(1, it->v(1) + _sampling.at(cid_ub).mom_corr.at(1).at(yPosID) );
-			//~ it->setv(2, it->v(2) + _sampling.at(cid_ub).mom_corr.at(2).at(yPosID) );
+			for (const auto d : _directions) {
+				it->setv(d, it->v(d) + _sampling.at(cid_ub).velo_corr.at(d).at(yPosID) ); 
+			}
 		}
 	}
 	
 	// write to file
 	if(simstep % _control.freq.write == 0 && nRank == 0)
 	{
 		{
-			// construct filename
-			std::stringstream fnamestream;
-			fnamestream << "DriftCtrl_drift.dat";
-		
-			std::ofstream ofs(fnamestream.str().c_str(), std::ios::app);
-			std::stringstream outputstream;
-			//~ outputstream << "     simstep" << "                      vy" << "                      vz" << std::endl;
-		
-			outputstream << setw(12) << simstep;
-			for(uint32_t yPosID = 0; yPosID < _range.subdivision.numBins; ++yPosID)
-				outputstream << FORMAT_SCI_MAX_DIGITS << _sampling.at(_target.cid).momentum.at(1).global.at(yPosID);
-			outputstream << std::endl;
-			
-			ofs << outputstream.str();
+			const std::string fname = "DriftCtrl_drift.dat";
+			std::ofstream ofs;
+			ofs.open(fname, std::ios::app);
+			ofs << setw(12) << simstep;
+			for(uint32_t yPosID = 0; yPosID < _range.subdivision.numBins; ++yPosID) {
+				ofs << FORMAT_SCI_MAX_DIGITS << _sampling.at(_target.cid).velocity.at(1).global.at(yPosID);
+			}
```

Comment:
why not simply
```suggestion
			for(const auto &wtf : _sampling.at(_target.cid).velocity.at(1).global) {
				ofs << FORMAT_SCI_MAX_DIGITS << wtf;
			}
```
?

This simplification seems possible in several other locations too.

Reply:
Nice simplification! I used it two times. In the other cases in this file, the index, e.g. `yPosID`, is used for different vectors. Therefore, it was unfortunately not possible to apply the suggested method.